src/main/java/com/avail/interpreter/levelTwo/operation/L2_ENTER_L2_CHUNK.java

```diff
@@ -113,20 +130,91 @@ public void translateToJVM (
 		// :: if (!checkValidity()) {
 		translator.loadInterpreter(method);
 		translator.literal(method, offsetInDefaultChunk.value);
-		method.visitMethodInsn(
-			INVOKEVIRTUAL,
-			getInternalName(Interpreter.class),
-			"checkValidity",
-			getMethodDescriptor(
-				BOOLEAN_TYPE,
-				INT_TYPE),
-			false);
+		Interpreter.checkValidityMethod.generateCall(method);
 		final Label isValidLabel = new Label();
 		method.visitJumpInsn(IFNE, isValidLabel);
 		// ::    return null;
 		method.visitInsn(ACONST_NULL);
 		method.visitInsn(ARETURN);
 		// :: }
 		method.visitLabel(isValidLabel);
+
+		// Extract register values that were saved in a register dump by the
+		// corresponding L2_SAVE_ALL_AND_PC_TO_INT instruction, which nicely set
+		// up for us the lists of registers that were saved.  The interpreter
+		// should have extracted the registerDump for us already.
+		final @Nullable EnumMap<RegisterKind, List<Integer>>
+			localNumberLists =
+				translator.liveLocalNumbersByKindPerEntryPoint.get(instruction);
+		if (localNumberLists != null)
+		{
+			final List<Integer> boxedList =
+				localNumberLists.get(RegisterKind.BOXED);
+			final List<Integer> intsList =
+				localNumberLists.get(RegisterKind.INTEGER);
+			final List<Integer> floatsList =
+				localNumberLists.get(RegisterKind.FLOAT);
+			final int boxedCount = boxedList.size();
+			final int intsCount = intsList.size();
+			final int floatsCount = floatsList.size();
+
+			int countdown = boxedCount + intsCount + floatsCount;
+			if (countdown > 0)
+			{
+				// Extract the register dump from the current continuation.
+				translator.loadInterpreter(method);
+				getReifiedContinuationMethod.generateCall(method);
+				registerDumpMethod.generateCall(method);
+				// Stack now has the registerDump.
+				for (int i = 0; i < boxedCount; i++)
+				{
+					if (--countdown > 0)
+					{
+						method.visitInsn(DUP);
+						// Stack has two registerDumps if needed.
+					}
+					translator.intConstant(method, i + 1);  //one-based
+					extractObjectAtMethod.generateCall(method);
+					method.visitVarInsn(
+						RegisterKind.BOXED.storeInstruction,
+						boxedList.get(i));
+				}
+				int i;
+				for (i = 0; i < intsCount; i++)
+				{
+					if (--countdown > 0)
+					{
+						method.visitInsn(DUP);
+						// Stack has two registerDumps if needed.
+					}
+					translator.intConstant(method, i + 1);  //one-based
+					extractLongAtMethod.generateCall(method);
+					method.visitInsn(L2I);
+					method.visitVarInsn(
+						RegisterKind.INTEGER.storeInstruction,
+						intsList.get(i));
+				}
+				for (int j = 0; j < floatsCount; j++, i++)
+				{
+					if (--countdown > 0)
+					{
+						method.visitInsn(DUP);
+						// Stack has two registerDumps if needed.
+					}
+					translator.intConstant(method, i + 1);  //one-based
+					extractLongAtMethod.generateCall(method);
+					method.visitInsn(L2D);
```

Comment:
Ah, this is the other side of the encoding bug in `L2PcOperand`. `L2D` isn't a reinterpret cast, so you'll need to use `Double.longBitsToDouble` here (after fixing the other side by changing `D2L` into `Double.doubleToRawLongBits`).

Reply:
Fixed.